File: webserver/var/srv_cgi/cgi-bin/calculate_mean.py
import sys


def parse_arguments(query_string):
    params = query_string.split('&')
    parsed_args = []
    for param in params:
        try:
            parsed_args.append(int(param))
        except ValueError:
            pass
    return parsed_args

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Error: one argument is needed.")
        sys.exit(1)

    query_string = sys.argv[1]
    parsed_args = parse_arguments(query_string)

    if parsed_args:
        mean = sum(parsed_args) / len(parsed_args)
        print("The mean of the integer values is: ", int(mean))
    else:
        print("There are no integer values to calculate the mean.")

File: webserver/var/srv_cgi/cgi-bin/calculate_mean.php
<?php

function parse_arguments($query_string) {
    $params = explode('&', $query_string);
    $parsed_args = [];
    foreach ($params as $param) {
        if (is_numeric($param)) {
            $parsed_args[] = intval($param);
        }
    }
    return $parsed_args;
}

if ($argc < 2) {
    echo "Error: one argument is needed.\n";
    exit(1);
}

$query_string = $argv[1];
$parsed_args = parse_arguments($query_string);

if (!empty($parsed_args)) {
    $mean = array_sum($parsed_args) / count($parsed_args);
    echo "The mean of the integer values is: ", intval($mean), "\n";
} else {
    echo "There are no integer values to calculate the mean.\n";
}
?>
